site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,296 @@
+      - helmTemplate: {}
+      deploy:
+      - kapp: {}
+EOF
+```
+
+**3. Create Package Repository**: A package repository is a collection of packages and their metadata. We will use `imgpkg` bundle to create package repository.
+
+For the purpose of this tutorial, we will run an unsecured local docker registry. In the real world please be safe and use appropriate security measures.
+    
+```bash
+$ docker run -d -p 5000:5000 --restart=always --name registry registry:2
+```
+
+From the terminal we can access this registry as localhost:5000 but within the cluster we'll need the IP Address. We will store the IP address in a variable:
+
+```bash 
+$ export REPO_HOST="`ifconfig | grep inet | grep -E '\b10\.' | awk '{ print $2}'`:5000"
+```
+
+Confirm that REPO_HOST is set to <IP_ADDRESS:PORT>
+
+```bash
+$ echo $REPO_HOST
+  10.104.3.7:5000
+```
+
+Lets start by creating the needed directories:
+
+```bash
+$ mkdir -p nginx-bitnami-repo nginx-bitnami-repo/.imgpkg nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+We can copy our CR YAMLs from the previous step in to the proper packages subdirectory. 
+
+```bash
+$ cp 1.0.0.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+$ cp pkgMetadata.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+Now, let’s use kbld to record which package bundles are used:
+
+```bash
+$ kbld -f nginx-bitnami-repo/packages/ --imgpkg-lock-output nginx-bitnami-repo/.imgpkg/images.yml
+```
+
+We will push the bundle onto repository using `imgpkg`.
+
+```bash
+$ imgpkg push -b ${REPO_HOST}/packages/nginx-bitnami-repo:1.0.0 -f nginx-bitnami-repo
+```
+
+We can verify by checking the Docker registry catalog:
+
+```bash
+$ curl ${REPO_HOST}/v2/_catalog
+```
+
+--------------------------
+
+## Consuming Carvel Helm Package:
+
+**1. Install Package Repository**: Before installing a package, we have to create a PackageRepository first. A PackageRepository is a collection of packages which are available to install. 
+```bash
+$ cat > repo.yml << EOF
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageRepository
+metadata:
+  name: simple-package-repository
+spec:
+  fetch:
+    imgpkgBundle:
+      image: ${REPO_HOST}/packages/nginx-bitnami-repo:1.0.0
+EOF
+```
+
+```bash
+$ kapp deploy -a repo -f repo.yml -y
+```
+
+After deploying, wait for the packageRepository description to become `Reconcile succeeded`. 
+
+```bash 
+$ kubectl get packagerepository
+```
+
+Now, we can see the list of package metadata's available
+
+```bash
+$ kubectl get packagemetadatas
+```
+
+**2. List Packages**:
+We can see the list of packages and their version's available for install 
+
+```bash 
+kubectl get packages
+```
+
+As we can see, our published nginx helm package is available for us to install.
+
+**3. Create Service Account**: To install the above package, we need to create default-ns-sa service account that give kapp-controller privileges to create resources in the default namespace
```

Comment:
"that give kapp-controller privileges" => that give PackageInstall CR privileges